homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,151 @@
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialise the config flow."""
+        super().__init__()
+        self._errors = {}
+
+    async def _test_connection(self, info):
+        """Verify that the camera data is valid before we add it."""
+        await self.async_set_unique_id(info[CONF_STILL_IMAGE_URL])
+        try:
+            self._abort_if_unique_id_configured()
+        except data_entry_flow.AbortFlow as err:
+            if "already_configured" in str(err):
+                self._errors["base"] = "still_image_already_configured"
+                return False
+        cam = GenericCamera(self.hass, info)
+        # Minimum functionality is getting a still image
+        image = await cam.async_camera_image()
+        if image is None:
+            self._errors["base"] = "unable_still_load"
+            return False
+        fmt = imghdr.what(None, h=image)
+        _LOGGER.debug(
+            "Still image at '%s' detected format: %s", info[CONF_STILL_IMAGE_URL], fmt
+        )
+        if fmt is None:
+            self._errors["base"] = "invalid_still_image"
+            return False
+
+        # Second level functionality is to get a stream.
+        if info.get(CONF_STREAM_SOURCE) not in [None, ""]:
+            try:
+                container = await self.hass.async_add_executor_job(
+                    av.open, info.get(CONF_STREAM_SOURCE)
+                )
+                video_stream = container.streams.video[0]
+                if video_stream is not None:
+                    return True
+            except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+                self._errors["base"] = "stream_unauthorised"
+            except (KeyError, IndexError):
+                self._errors["base"] = "stream_novideo"
+            except OSError as err:
+                if "No route to host" in str(err):
+                    self._errors["base"] = "stream_no_route_to_host"
+            return False
+        return True
+
+    async def async_step_user(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Handle the start of the config flow."""
+
+        self._errors = {}
+
+        if user_input is not None:
+            if await self._test_connection(user_input):
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_AUTHENTICATION] = HTTP_BASIC_AUTHENTICATION
+            user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE] = False
+            user_input[CONF_CONTENT_TYPE] = DEFAULT_CONTENT_TYPE
+            user_input[CONF_FRAMERATE] = 2
+            user_input[CONF_VERIFY_SSL] = True
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
```

Comment:
This would be more readable if you built the schema in a separate function

Reply:
Done at 1ec108d